src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,182 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final int SETTING = 0;
+    private static final int NOT_SETTING = 1;
+    private int size;
+    private transient Node<T> first;
+    private transient Node<T> last;
+
     @Override
     public void add(T value) {
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkPositionIndex(index, NOT_SETTING);
+        if (index == size) {
+            linkLast(value);
+        } else {
+            linkBefore(value, node(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkPositionIndex(index, SETTING);
+        return node(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkPositionIndex(index, SETTING);
+        Node<T> node = node(index);
+        T oldValue = node.item;
+        node.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkPositionIndex(index, SETTING);
+        return unlink(node(index));
     }
 
     @Override
     public boolean remove(T object) {
+        if (object == null) {
+            for (Node<T> currentNode = first; currentNode != null; currentNode = currentNode.next) {
+                if (currentNode.item == null) {
+                    unlink(currentNode);
+                    return true;
+                }
+            }
+        } else {
+            for (Node<T> currentNode = first; currentNode != null; currentNode = currentNode.next) {
+                if (object.equals(currentNode.item)) {
+                    unlink(currentNode);
+                    return true;
+                }
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        if (first == null && last == null) {
+            return true;
+        }
+        return size == 0;
+    }
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(Node<T> prev,T element,Node<T> next) {
+            this.next = next;
+            this.prev = prev;
+            this.item = element;
+        }
+
     }
+
+    private void linkLast(T element) {
+        final Node<T> prev = last;
+        final Node<T> newNode = new Node<>(prev,element,null);
+        last = newNode;
+        if (prev == null) {
+            first = newNode;
+        } else {
+            prev.next = newNode;
+        }
+        size++;
+    }
+
+    private void checkPositionIndex(int index, int expressionToCheck) {
```

Comment:
I suggest just calling method isPositionIndex (and rename to checkIndex) and isPositionIndexToSet (rename to checkIndexOnSet). In this case you should also add exception throwing logic to checkIndex and checkIndexOnSet methods directry it will reduce amount of code, complexity and constants